pkg/ruler/mapper.go

```diff
@@ -156,11 +156,9 @@ func (m *mapper) writeRuleGroupsIfNewer(groups []rulefmt.RuleGroup, filename str
 		if err != nil {
 			return false, err
 		}
-		newHash := md5.New()
-		currentHash := md5.New()
 
 		// bailout if there is no update
-		if string(currentHash.Sum(current)) == string(newHash.Sum(d)) {
+		if bytes.Equal(current, d) {
```

Comment:
A quick benchmark on a medium-sized rule group file says a direct bytes compare is inconsequential in terms of timing:

```
BenchmarkRuleGroupBytesCompare-16                 	  237231	     50473 ns/op
BenchmarkRuleGroupBytesCompareAlreadyEqual-16     	  266635	     44484 ns/op
BenchmarkRuleGroupMD5Compare-16                   	  226880	     52006 ns/op
BenchmarkRuleGroupMD5CompareAlreadyEqual-16       	  263240	     44985 ns/op
BenchmarkRuleGroupSHA256Compare-16                	  228654	     51549 ns/op
BenchmarkRuleGroupSHA256CompareAlreadyEqual-16    	  262724	     44853 ns/op
```
IMO `bytes.Equal` is more expressive, the code says what it's trying to do.